### PR TITLE
bug/MOBILE-3507 missing transaction event from feed

### PR DESCRIPTION
**Pull request by agustk**
A situation can occur where transaction events are far removed from their transaction and also have a totally different date as their transaction. To make the app easier to use a feature in the sdk is best that makes sure the events are next to their transactions.

### CODE REVIEW

File: sdk/src/main/java/com/meniga/sdk/models/feed/MenigaFeed.java [hunk 1]
```diff
@@ -8,12 +8,14 @@
...
 import com.meniga.sdk.helpers.Result;
 import com.meniga.sdk.models.feed.operators.MenigaFeedOperations;
+import com.meniga.sdk.models.transactions.MenigaTransaction;
 
 import org.joda.time.DateTime;
 
 import java.io.Serializable;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.stream.Stream;
```
Comment (arturdm) [inline sdk/src/main/java/com/meniga/sdk/models/feed/MenigaFeed.java]: This import is not needed.

File: sdk/src/main/java/com/meniga/sdk/models/feed/MenigaFeed.java [hunk 2]
```diff
@@ -73,6 +75,29 @@ public void writeToParcel(Parcel dest, int flags) {
 		dest.writeList(this);
 	}
 
+	public void moveTransactionEventsToTransaction() {
```
Comment (arturdm) [inline sdk/src/main/java/com/meniga/sdk/models/feed/MenigaFeed.java]: If the library client does not have to perform this operation maybe we should keep it in the converter?

File: sdk/src/main/java/com/meniga/sdk/converters/MenigaFeedConverter.kt [hunk 3]
```diff
@@ -28,6 +28,8 @@ class MenigaFeedConverter(private val feedItemFactory: FeedItemFactory) : Meniga
                         add(feedItemFactory.getMenigaFeetItem(it as JsonObject))
                     }
 
+                    moveTransactionEventsToTransaction();
```
Comment (arturdm) [inline sdk/src/main/java/com/meniga/sdk/converters/MenigaFeedConverter.kt]: You can use kotlin list operations here:

```
                    val transactionEvents = (0 .. size).map { get(it) }
                            .filterIsInstance(MenigaTransactionEvent::class.java)
                    
                    removeAll { it is MenigaTransactionEvent }
                    
                    transactionEvents.forEach { 
                        ... and so on
                    }
```

File: sdk/src/main/java/com/meniga/sdk/models/feed/MenigaFeed.java (hunk 2 above)
Comment (agustk) [reply sdk/src/main/java/com/meniga/sdk/models/feed/MenigaFeed.java]: True, good point

File: sdk/src/main/java/com/meniga/sdk/models/feed/MenigaFeed.java (hunk 1 above)
Comment (agustk) [reply sdk/src/main/java/com/meniga/sdk/models/feed/MenigaFeed.java]: Nope, removing

**Pull request merged**